### PR TITLE
Make JIT Aliasing Test Less Brittle

### PR DESCRIPTION
**Pull request by alanwaketan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #65493

Summary:
Added a last resolve to use whatever ATen operator that has Tensor outputs in the graph as the operator node to check alias annotation.

Test Plan:
python test/test_ops.py -k test_variant_consistency_jit

Differential Revision: [D31321221](https://our.internmc.facebook.com/intern/diff/D31321221)

### CODE REVIEW

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31132861).

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -9542,9 +9535,7 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
                # RuntimeError:
                # Arguments for call are not valid.
                DecorateInfo(unittest.skip("Skipped!"), 'TestJit', 'test_variant_consistency_jit', dtypes=(torch.complex64,)),
-               # RuntimeError: aliasOp != torch::jit::getOperatorAliasMap().end()
-               # INTERNAL ASSERT FAILED at "../torch/csrc/jit/passes/utils/check_alias_annotation.cpp":157,
-               # please report a bug to PyTorch.
+               # RuntimeError: Arguments for call are not valid.
```
Comment (eellison) [inline torch/testing/_internal/common_methods_invocations.py:9538]: What are the new errors here ? Do you know why this is happening ?

**Review by eellison — COMMENTED** [review]
looks good!! one question

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (alanwaketan) [reply torch/testing/_internal/common_methods_invocations.py:9538]: Here is the full message:
```
RuntimeError: 
Arguments for call are not valid.
The following variants are available:
  
  aten::norm.Scalar(Tensor self, Scalar p=2) -> (Tensor):
  Expected a value of type 'number' for argument 'p' but instead found type 'str'.
  
  aten::norm.ScalarOpt_dim(Tensor self, Scalar? p, int[1] dim, bool keepdim=False) -> (Tensor):
  Expected a value of type 'Optional[number]' for argument 'p' but instead found type 'str'.
  
  aten::norm.names_ScalarOpt_dim(Tensor self, Scalar? p, str[1] dim, bool keepdim=False) -> (Tensor):
  Expected a value of type 'Optional[number]' for argument 'p' but instead found type 'str'.
  
  aten::norm.ScalarOpt_dim_dtype(Tensor self, Scalar? p, int[1] dim, bool keepdim, *, int dtype) -> (Tensor):
  Expected a value of type 'Optional[number]' for argument 'p' but instead found type 'str'.
  
  aten::norm.dtype_out(Tensor self, Scalar? p, int[1] dim, bool keepdim, *, int dtype, Tensor(a!) out) -> (Tensor(a!)):
  Expected a value of type 'Optional[number]' for argument 'p' but instead found type 'str'.
  
  aten::norm.out(Tensor self, Scalar? p, int[1] dim, bool keepdim=False, *, Tensor(a!) out) -> (Tensor(a!)):
  Expected a value of type 'Optional[number]' for argument 'p' but instead found type 'str'.
  
  aten::norm.ScalarOpt_dtype(Tensor self, Scalar? p, *, int dtype) -> (Tensor):
  Expected a value of type 'Optional[number]' for argument 'p' but instead found type 'str'.
  
  aten::norm.names_ScalarOpt_dim_dtype(Tensor self, Scalar? p, str[1] dim, bool keepdim, *, int dtype) -> (Tensor):
  Expected a value of type 'Optional[number]' for argument 'p' but instead found type 'str'.
  
  aten::norm.names_dtype_out(Tensor self, Scalar? p, str[1] dim, bool keepdim, *, int dtype, Tensor(a!) out) -> (Tensor(a!)):
  Expected a value of type 'Optional[number]' for argument 'p' but instead found type 'str'.
  
  aten::norm.names_out(Tensor self, Scalar? p, str[1] dim, bool keepdim=False, *, Tensor(a!) out) -> (Tensor(a!)):
  Expected a value of type 'Optional[number]' for argument 'p' but instead found type 'str'.

The original call is:
  File "<string>", line 3

def the_method(i0):
    return i0.norm('fro')
           ~~~~~~~ <--- HERE
```

I suspect there is a mismatch between the passed in operator and the TS node.

**Review by eellison — COMMENTED** [review]
Sorry I'm going to ask one more thing - do you mind just removing all of the `test_variant_consistency_jit` skips, and seeing what test failures remain ? I'm pretty sure that many of the skips here, such as `nn.functional.adaptive_avg_pool2d`, will be fixed with this PR

**Comment by alanwaketan** [comment]
> Sorry I'm going to ask one more thing - do you mind just removing all of the `test_variant_consistency_jit` skips, and seeing what test failures remain ? I'm pretty sure that many of the skips here, such as `nn.functional.adaptive_avg_pool2d`, will be fixed with this PR

Sure, let me try. I will post the results soon here.

**Comment by alanwaketan** [comment]
> > Sorry I'm going to ask one more thing - do you mind just removing all of the `test_variant_consistency_jit` skips, and seeing what test failures remain ? I'm pretty sure that many of the skips here, such as `nn.functional.adaptive_avg_pool2d`, will be fixed with this PR
> 
> Sure, let me try. I will post the results soon here.

Here is the raw output. Will process that later on today.
[output2.txt](https://github.com/pytorch/pytorch/files/7238122/output2.txt)

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31132861).

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31132861).

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31132861).

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31132861).

**Review by eellison — APPROVED** [review]
LGTM !! Thank you for all the work here.

**Comment by alanwaketan** [comment]
> LGTM !! Thank you for all the work here.

Thanks for approving this patch.

**Comment by suo** [comment]
this broke master, reverting: https://hud.pytorch.org/commit/9f97c66a7a62e0f9bf5296adb6c4086efbf69ade

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31321221).

**Pull request closed without merging**